### PR TITLE
Read/write TableList

### PR DESCRIPTION
**Pull request by nstarman**
Allow TableList I/O, using ASDF. Tables may be written one of two ways: as a single ASDF object or as separate files (anything supported by Table) with a coordinating container.

### CODE REVIEW

**Comment by bsipocz** [comment]
@nstarman - Thanks for the PR. However, I'm afraid without a very strong use case I don't think this is appropriate for the internal machinery. ASDF while might be the preference of some users, it's not yet the format of choice by the archives distribute their data, neither it's a standard table list in the wider ecosystem.

**Comment by nstarman** [comment]
Using json as the coordinating object would be an easy adaptation

**Comment by keflavich** [comment]
`TableLists` are presently used only by `Vizier`, right?  Or do we have other services that use it?  (if you happen to know - I'm just pulling on memory)

I do like the idea of having a way to load and save these files, but we generally don't include io machinery in astroquery - this might actually be more appropriate for the core package.

That said, the TableList object is also something we invented here... so it might be appropriate.

**Comment by nstarman** [comment]
It looks like the module ``esasky`` also imports `TableList`

File: astroquery/utils/commons.py [hunk 1]
```diff
@@ -297,6 +310,220 @@ def pprint(self, **kwargs):
...
+
+    @classmethod
+    def read(
+        cls,
+        drct: str,
+        format: T.Union[str, T.Sequence] = None,
+        suffix: T.Optional[str] = None,
+        **table_kw,
+    ):
+        """Write to ASDF.
```
Comment (keflavich) [inline astroquery/utils/commons.py:457]: this is the reader

**Comment by keflavich** [comment]
Just the io.  But this really raises questions: `TableList` is kind of a hack to make display of lists of tables a little nicer.  This functionality is useful, but it serves to highlight that we have a weird format in astroquery.

That said, since this format _is_ unique to astroquery at the moment, it's not crazy to have this tool here.

I think I'm happy to include it as long as we keep it clear that `asdf` is an optional dependency.

However, before we go down this route, since you've been investigating this @nstarman : might there be something that already has io functionality that we could use to replace the `TableList` object?  I don't have anything in mind, but I think we'd generally prefer to use externally defined (and maintained) data structures than to develop our own sophisticated ones with custom serialization formats.

**Comment by bsipocz** [comment]
I'm still 👎  on this. There is no clear user demand for this that would guarantee the functionality to live in astroquery rather than in your own code (I don't see any feature requests raised over the years), and this data scructure can certainly a place for future cleanup of astroquery to use something more widely used, rather than the "ship our own". If we keep the current `TableList`, then I keep looking at it as a convenienc layer on top of Table, and not something that needs more paching up.

**Comment by bsipocz** [comment]
Thank you for the effort in this PR. There are both conflicts and unresolved discussions above which lean to the direction that extending the hack that `TableList` is not exactly desirable, so I'm going to close this now.

**Pull request closed without merging**